### DISCUSSION TITLE
UBI backend and tools with multiple files to extract like mold

### DISCUSSION THREAD

**Discussion by pdecat** [jdx/mise#2951] [Q&A]
mold support was added with the [UBI](https://github.com/houseabsolute/ubi) backend in https://github.com/jdx/mise/pull/2792

Given UBI only does single binary installs, what's the proper way to get the `ld.mold` symlink to the `mold` binary which is needed for tools like clang for linking?

Right now, I do this:
```
# mise use -g mold
# ln -snf $(which mold) ~/.local/bin/ld.mold
```

Using the [`exe` tool option](https://mise.jdx.dev/dev-tools/backends/ubi.html#exe) like the following does not work as it is a symlink in the [tar archive](https://github.com/rui314/mold/releases/download/v2.34.1/mold-2.34.1-x86_64-linux.tar.gz) :
```
[tools]
"ubi:rui314/mold" = { version = "latest", exe = "ld.mold" }
```
=>
```
# mise install
mise ERROR Failed to install ubi:rui314/mold@2.34.1
mise ERROR Failed to install with ubi 'ubi:rui314/mold@2.34.1': could not find any files named ld.mold in the downloaded tarball
mise ERROR Run with --verbose or MISE_VERBOSE=1 for more information
```

FWIW, I stumbled on this issue which may help in the future if implemented: https://github.com/houseabsolute/ubi/issues/68

Somewhat related to:
* https://github.com/jdx/mise/issues/2875
* https://github.com/jdx/mise/issues/2885

Edit: switched to aqua backend with `"aqua:rui314/mold" = "latest"`

**Comment by jdx** [comment]
I think what would be helpful is if ubi could extract the entire tarball rather than just a single bin, maybe a `ubi --all` flag or something

**Comment by Marukome0743** [comment]
The practical installation issue described here can now be avoided by using the current `mold` registry entry, which installs through the Aqua backend rather than UBI:

```sh
mise use mold
# or explicitly:
mise use aqua:rui314/mold
```

This distinction matters because UBI installs a selected executable and does not preserve the complete archive layout. The Aqua package installs mold’s full release archive, including the `ld.mold` symlink.

I verified this in a clean Linux arm64 container using mise 2026.7.18. It installed mold 2.41.0, and both commands worked:

```sh
mold --version
ld.mold --version
```

The installed files included:

```text
ld.mold -> mold
```

Both commands reported mold 2.41.0.

The UBI backend itself has since been deprecated, and its registry entries were removed in #7374, released in mise v2025.12.11. Therefore, new configurations should use the `mold` shorthand or the explicit Aqua backend rather than `ubi:rui314/mold`.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
